include/ruby/internal/intern/string.h

```diff
@@ -601,6 +601,19 @@ VALUE rb_str_dup(VALUE str);
  */
 VALUE rb_str_resurrect(VALUE str);
 
+/**
+ * Returns whether a string is chilled or not.
+ *
+ * This function is temporary and users must check for its presence using
+ * mkmf's have_func helper. If this function is not implemented, then strings
+ * can't be chilled.
+ *
+ * @param[in]  str  A string.
+ * @retval     1    The string is chilled.
+ * @retval     0    Otherwise.
+ */
+int rb_str_chilled_p(VALUE str);
```

Comment:
We could use `bool` as a return type these days, e.g. there is already `include/ruby/vm.h:57:bool ruby_free_at_exit_p(void);` and more.

Reply:
Ah, I wasn't sure. Updated.